pkgs/desktops/gnome-3/extensions/buildGnomeExtension.nix

```diff
@@ -0,0 +1,43 @@
+{ pkgs, lib, stdenv, fetchurl, gnome3, unzip }:
+
+let
+  buildGnomeExtension = {
+    # Every gnome extension has a UUID. It's the name of the extension folder once unpacked
+    # and can always be found in the metadata.json of every extension.
+    uuid,
+    name,
+    pname,
+    description,
+    # extensions.gnome.org extension URL
+    link,
+    version,
+    sha256,
+  }:
+
+  stdenv.mkDerivation {
+    inherit pname;
+    version = builtins.toString version;
+    src = fetchurl {
+      url = "https://extensions.gnome.org/extension-data/${
+          builtins.replaceStrings [ "@" ] [ "" ] uuid
+        }.v${builtins.toString version}.shell-extension.zip";
+      inherit sha256;
+    };
+    nativeBuildInputs = [ unzip ];
+    buildCommand = ''
+      mkdir -p $out/share/gnome-shell/extensions/
+      unzip $src -d $out/share/gnome-shell/extensions/${uuid}
+    '';
+    meta = {
+      description = builtins.head (lib.splitString "\n" description);
+      longDescription = description;
+      homepage = link;
+      license = lib.licenses.gpl2Plus; # https://wiki.gnome.org/Projects/GnomeShell/Extensions/Review#Licensing
+      extensionUuid = uuid; # Store the extension's UUID, because we might need it at some places
+      maintainers = with lib.maintainers; [ piegames ];
+    };
+    # You don't need a remote machine for unzipping a bit of JavaScript
```

Comment:
I think this isn't a great idea. If I already have all of the unpacking done on my CI machine why would I prefer to do it locally? Is this solving an actual problem or just "because we can"?

Reply:
I thought it was good practice to reduce overhead by making trivial builds run locally. But I can change it back.

Reply:
I think preferLocalBuild only makes sense for packages where we expect normal users to have a cache miss, right?
Anyways I guess it only makes really sense for "leaves" in the derivation, right? Otherwise you need to upload it to a build server again, when you build the next dependency?
I’m unsure. @andir your concern is about the situation where you use a remote host (who already has this package build) as a remote builder but not as substituter right? Because as far as I understand packages with preferLocalBuild will still be substituted.
I think in the end it just doesn‘t matter much because users will get this cached from hydra …

Reply:
My concern isn't for a particular use case but because there is none that justifies this. If we do not have a reason to do it we shouldn't.